trieste/acquisition/function/active_learning.py

```diff
@@ -0,0 +1,234 @@
+This module contains acquisition function builders and acquisition functions for Bayesian active
+learning.
+"""
+
+from __future__ import annotations
+
+from typing import Optional
+
+import tensorflow as tf
+import tensorflow_probability as tfp
+
+from ...data import Dataset
+from ...models import ProbabilisticModel
+from ...types import TensorType
+from ...utils import DEFAULTS
+from ..interface import AcquisitionFunction, SingleModelAcquisitionBuilder
+
+
+class PredictiveVariance(SingleModelAcquisitionBuilder):
+    """
+    Builder for the determinant of the predictive covariance matrix over the batch points.
+    For a batch of size 1 it is the same as maximizing the predictive variance.
+    """
+
+    def __init__(self, jitter: float = DEFAULTS.JITTER) -> None:
+        """
+        :param jitter: The size of the jitter to use when stabilising the Cholesky decomposition of
+            the covariance matrix.
+        """
+        super().__init__()
+
+        self._jitter = jitter
+
+    def __repr__(self) -> str:
+        """"""
+        return f"PredictiveVariance(jitter={self._jitter!r})"
+
+    def prepare_acquisition_function(
+        self,
+        model: ProbabilisticModel,
+        dataset: Optional[Dataset] = None,
+    ) -> AcquisitionFunction:
+        """
+        :param model: The model.
+        :param dataset: Unused.
+
+        :return: The determinant of the predictive function.
+        """
+
+        return predictive_variance(model, self._jitter)
+
+    def update_acquisition_function(
+        self,
+        function: AcquisitionFunction,
+        model: ProbabilisticModel,
+        dataset: Optional[Dataset] = None,
+    ) -> AcquisitionFunction:
+        """
+        :param function: The acquisition function to update.
+        :param model: The model.
+        :param dataset: Unused.
+        """
+        return function  # no need to update anything
+
+
+def predictive_variance(model: ProbabilisticModel, jitter: float) -> TensorType:
+    """
+    The predictive variance acquisition function for active learning, based on
+    the determinant of the covariance (see :cite:`MacKay1992` for details).
+    Note that the model needs to supply covariance of the joint marginal distribution,
+    which can be expensive to compute.
+
+    :param model: The model of the objective function.
+    :param jitter: The size of the jitter to use when stabilising the Cholesky decomposition of
+            the covariance matrix.
+    """
+
+    @tf.function
+    def acquisition(x: TensorType) -> TensorType:
+
+        try:
+            _, covariance = model.predict_joint(x)
+        except NotImplementedError:
+            raise ValueError(
+                """
+                PredictiveVariance only supports models with a predict_joint method.
+                """
+            )
+        return tf.exp(tf.linalg.logdet(covariance + jitter))
+
+    return acquisition
+
+
+class ExpectedFeasibility(SingleModelAcquisitionBuilder):
```

Comment:
name comes from Bichon paper, Ranjan did not name theirs really, so as a shortcut I used this name for both which seems appropriate given that criteria are very related - it also seemed better than something like `BichonRanjanCriterion`

Reply:
Agreed, although expected feasibility is possibly misleading? Sounds like a probability of feasibility, which could be confusing.

Reply:
I think I would keep it as it is, many of them sound similar, you have to check the function itself to see what is it for